src/GslCore/DnaCreation.fs

```diff
@@ -552,7 +515,28 @@ let expandAssembly
 
                 match ppp.part with
                 | MARKERPART ->
-                    yield expandMarkerPart library dnaSource ppp
+                    // Choose provider
+                    let providers =
+                        markerProviders |> 
+                            List.choose (fun provider -> 
+                                                    match provider.ScoreJob a.capabilities with
+                                                        | None -> None
+                                                        | Some(score) -> Some (score,provider)
+                                        ) 
+                    if providers = [] then failwithf "MARKERPART substitution, no marker provider available"
+                    let _,markerProvider = providers |> List.maxBy (fst)
+
+                    let markerSet =
+                        match a.pragmas.TryGetOne("markerset") with
+                            | Some(x) ->
+                                let x' = x.ToLower()
+                                if not (markerProvider.IsLegal x') then
+                                    failwithf "ERROR: unknown marker set %s, expected one of %s\n"
+                                        x' (String.Join(",",markerProvider.ListMarkers()))
+                                else x'
+                            | None -> "default"

```

Comment:
Hmm, this seems odd.  Rather than using the in-band magic string "default" to indicate use of the default marker set, why not allow the provider to accept Option<string> where None implies the default case?